src/app/sourceHighlighter.js

```diff
@@ -0,0 +1,206 @@
+var utils = require('./utils')
+var remix = require('ethereum-remix')
+var ace = require('brace')
+var Range = ace.acequire('ace/range').Range
+
+/*
+  Provides source highlighting when debugging a transaction
+*/
+function SourceHighlighter (editor, txdebugger, compilerEvent, appEvent, switchToFile) {
+  this.switchToFile = switchToFile
+  this.editor = editor
+  this.txdebugger = txdebugger
+  this.sourceMappingDecoder = new remix.util.SourceMappingDecoder()
+  this.compilationData
+
+  this.currentSourceMap
+  this.currentLineColumnLayout // used to retrieve line/column from char/length
+  this.currentRange
+  this.currentMarker
+  this.currentExecutingAddress
+  this.currentFile
+
+  this.isDebugging = false
+
+  var self = this
+
+  /* hide/show marker when debugger does not have the focus */
+  appEvent.register('tabChanged', function (tab) {
+    if (tab !== 'debugView') {
+      self.editor.removeMarker(self.currentMarker)
+    } else {
+      if (self.currentRange) {
+        self.currentMarker = self.editor.addMarker(self.currentRange, 'highlightcode')
+      }
+    }
+  })
+
+  /* update compilation data */
+  compilerEvent.register('compilationFinished', this, function (success, data, source) {
+    if (!self.isDebugging) {
```

Comment:
Do we really want this check? Don't we usually start debugging _after_ the compilation finished?